lib/api/dsl/methods.js

```diff
@@ -109,6 +109,72 @@ var geoUtil = {
   },
 
   /**
+   * Construct a valid usable BBox
+   *
+   * @param {Object} the given object
+   * @return {Object} the valid usable BBox object
+   */
+  constructPolygon: function(geoFilter) {
+    var i,
+      tmp,
+      point,
+      entry,
+      polygon = [];
+
+    if (geoFilter.points === undefined) {
+      throw new BadRequestError('No point list found');
+    }
+
+    if (!_.isArray(geoFilter.points)) {
+      throw new BadRequestError('A polygon must be in array format');
+    }
+
+    if (geoFilter.points.length < 3) {
+      throw new BadRequestError('A polygon must have at least 3 points');
+    }
+
+    /*eslint guard-for-in: 0 */
```

Comment:
I don't think we need an exception for eslint here.

We can add a if.. hasOwnProperty as suggested or use an alternative to for..in like Array.prototype.forEach